wire.md

```diff
@@ -0,0 +1,1008 @@
+
+A responder SHOULD include the hashes of all known `post/text` and
+`post/delete` posts made to a channel between `time_start` and `time_end`.
+
+A `time_end` of 0 MUST be understood as a request to keep this request alive
+even after all known hashes in the range `time_start` to `now()` are provided,
+and continue to receive any new chat messages that the responder learns of in
+the future, so long as this request is still alive. `time_start` MUST be
+respected regardless of whether `time_end` is 0 or some definite value.
+
+Responding clients SHOULD respond with all known chat messages within the
+requested time range, though they may desire not to in certain circumstances,
+particularly if a channel has a very long history and the responding client
+lacks sufficient resources at the time to return thousands or hundreds of
+thousands of chat message hashes.
+
+A `limit` of 0 MUST be understood as having no maximum on the number of hashes
+the requester wishes to receive.
+
+##### 6.3.2.5 Channel State Request
+
+Request posts that describe the current state of a channel and its members, and
+optionally subscribe to future state changes.
+
+field          | type               | desc
+---------------|--------------------|-----------------------------------
+`channel_len`  | `varint`           | length of the channel's name, in bytes 
+`channel`      | `u8[channel_len] ` | channel name (UTF-8)
+`future`       | `varint`           | whether to include live / future state hashes
+
+`msg_type` MUST be set to `5`.
+
+A responder receiving this request MUST respond with 1 or more Hash Response
+messages, with only posts that relate to the current state of the channel.
+Requesters MAY discard hashes mapping to posts that do not contain relevant
+information.
+
+See Section 5.4.4 for context on what comprises channel state. Chat messages
+MUST NOT be included in responses to this request. Clients MUST be able to
+handle the hashes of unexpected types appearing in responses, and MAY choose
+for themselves whether to discard them or not.
+
+`future` MUST be set to either `1` or `0`.
+
+If `future = 1`, the responder SHOULD respond with future channel state changes
+as they become known to the responder, and the request SHOULD be held open
+indefinitely on both the requester and responder side until a Cancel Request is
+issued by the requester, or the responder elects to end the request by sending
+a Hash Response with `hash_count = 0`.
+
+If `future = 1` and a post that is part of the latest state for a channel is
+deleted, the responder MUST immediately send the hash of the next-latest
+piece of state of that same type as a Hash Response. For example, if the latest
+`post/topic` setting the channel's topic string is deleted by its author with a
+`post/delete` post, the hash of the second-latest `post/topic` for that channel
+SHOULD be sent.
+
+If `future = 0`, only the latest state posts will be included, and the request
+MUST NOT be held open.
+
```

Comment:
Why is future boolean and not simply a `limit` int analogous to that of Channel Time Range Requests?

Reply:
I don't think a limit would be useful here without a means of paginating through the response set of hashes.